### PR TITLE
fix two flaky test -- TestP2PNoDuplicatedMessage and TestDeleteJobFromJobQueue

### LINKED ISSUE #1380: Fix TestP2PNoDuplicatedMessage.testP2PStateTransitionEnabled
LOG 1055 09/18 touch 6

> 2020-09-18T10:48:15.2643739Z [ERROR] testP2PStateTransitionEnabled(org.apache.helix.integration.messaging.TestP2PNoDuplicatedMessage)  Time elapsed: 24.341 s  <<< FAILURE!
2020-09-18T10:48:15.2688194Z java.lang.AssertionError: expected:<2200> but was:<2191>
2020-09-18T10:48:15.2691422Z 	at org.apache.helix.integration.messaging.TestP2PNoDuplicatedMessage.testP2PStateTransitionEnabled(TestP2PNoDuplicatedMessage.java:180)
2020-09-18T10:48:15.2694794Z 
2020-09-18T10:48:15.6628744Z [ERROR] Failures: 
2020-09-18T10:48:15.6634276Z [ERROR]   TestP2PNoDuplicatedMessage.testP2PStateTransitionEnabled:180 expected:<2200> but was:<2191>

### LINKED ISSUE #1406: fix TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue
LOG 1431

>2020-09-25T01:00:23.5925462Z [ERROR] testForceDeleteJobFromJobQueue(org.apache.helix.integration.task.TestDeleteJobFromJobQueue)  Time elapsed: 0.137 s  <<< FAILURE!
2020-09-25T01:00:23.5927674Z org.apache.helix.HelixException: Failed to delete job: testForceDeleteJobFromJobQueue_job2 from queue: testForceDeleteJobFromJobQueue
2020-09-25T01:00:23.5931032Z 	at org.apache.helix.integration.task.TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue(TestDeleteJobFromJobQueue.java:73)
2020-09-25T01:00:23.5933143Z 
2020-09-25T01:00:23.9833898Z [ERROR] Failures: 
2020-09-25T01:00:23.9836368Z [ERROR]   TestDeleteJobFromJobQueue.testForceDeleteJobFromJobQueue:73 » Helix Failed to ...
2020-09-25T01:00:23.9838007Z [ERROR] Tests run: 1204, Failures: 1, Errors: 0, Skipped: 1

### PR DESCRIPTION
**Pull request by kaisun2000**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

fix #1406; fix #1380 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Fix two flaky tests. For TestP2PNoDuplicatedMessage, cap the success
rate to 90 percent as in github environment ZK access can be slow.
For TestDeleteJobFromJobQueue, the design of force delete can be
in conflict with controller writes as contrary to what was noted by
the original deleveloper of this feature. We fix it by stop the
controller for now.
### Tests

- [ ] The following tests are written for this issue:

(List the names of added unit/integration tests)

- [ ] The following is the result of the "mvn test" command on the appropriate module:

(Before CI test pass, please copy & paste the result of "mvn test")

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java [hunk 1]
```diff
@@ -174,7 +174,10 @@ public void testP2PStateTransitionEnabled() {
       verifyP2PEnabled(startTime);
     }
 
-    Assert.assertEquals(p2pTrigged, total);
+    // The success rate really depends on how quick participant act in relationship with controller.
+    // For now, we set 90% threshold.
+    long threshold = Math.round(total * 0.9);
```
Comment (jiajunwang) [inline helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java:179]: @zhangmeng916 and @lei-xia , could you please confirm if this is the expectation? I understand that Kai needs to change it since the current run does not always succeed with 100% threshold. But I'm not sure if this should be our expectation?

Comment (kaisun2000) [reply helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java:179]: @jiajunwang, I sync-ed with Meng offline. @zhangmeng916, can you help to comment?

Comment (zhangmeng916) [reply helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java:179]: I think we cannot guarantee 100% P2P for all kinds of infrastructure. Although local tests always passed, there're some failures in our testing environment. The percent is always above 90% though. I believe there are corner cases that we cannot label all stale messages, which will block the sending of P2P.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java [hunk 2]
```diff
@@ -69,7 +69,14 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
     Assert
         .assertNotNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
 
-    // The following force delete for the job should go through without getting an exception
+    // Force deletion can have Exception thrown as controller is writing to propertystore path too.
+    // https://github.com/apache/helix/issues/1406, also force deletion may not be safe.
+    // Thus, we stop pipeline to make sure there is not such race condition.
+    _gSetupTool.getClusterManagementTool().enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(3000);
+    // note this sleep is critical as it would take time for controller to stop.
```
Comment (zhangmeng916) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:77]: Can you change to "TODO", and put before the sleep line?

File: helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java (hunk 1 above)
Comment (zhangmeng916) [reply helix-core/src/test/java/org/apache/helix/integration/messaging/TestP2PNoDuplicatedMessage.java:179]: Please make sure other reviewers are ok with this too.

**Review by zhangmeng916 — APPROVED** [review]
Please make sure other reviewers are ok too.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java (hunk 2 above)
Comment (kaisun2000) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:76]: changed.

**Comment by kaisun2000** [comment]
This PR is approved, please help to merge in.

>Fix two flaky tests. For TestP2PNoDuplicatedMessage, cap the success
rate to 90 percent as in github environment ZK access can be slow.
For TestDeleteJobFromJobQueue, the design of force delete can be
in conflict with controller writes as contrary to what was noted by
the original deleveloper of this feature. We fix it by stop the
controller for now.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java [hunk 3]
```diff
@@ -69,7 +69,14 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
     Assert
         .assertNotNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
 
-    // The following force delete for the job should go through without getting an exception
+    // Force deletion can have Exception thrown as controller is writing to propertystore path too.
+    // https://github.com/apache/helix/issues/1406, also force deletion may not be safe.
+    // Thus, we stop pipeline to make sure there is not such race condition.
+    _gSetupTool.getClusterManagementTool().enableCluster(CLUSTER_NAME, false);
```
Comment (pkuwm) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:75]: CLUSTER_NAME is a defined in the parent class which is also used by other tests. How could you ensure disabling this CLUSTER_NAME would not impact or fail other tests that use the same cluster?

Comment (kaisun2000) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:75]: `protected final String CLUSTER_NAME = CLUSTER_PREFIX + "_" + getShortClassName();` each test class has unique  cluster name.

Comment (pkuwm) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:75]: Ops, naming style as a constant but it is not... So easy to misread.

File: helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java [hunk 4]
```diff
@@ -69,7 +69,14 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
...
+    // note this sleep is critical as it would take time for controller to stop.
+    // todo: this is not best practice to sleep. Let GenericHelixController gives out a signal would
+    // todO: be another way. But this needs much more careful design.
```
Comment (jiajunwang) [inline helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:78]: Typo, and use uppercase "TODO", please.

**Review by jiajunwang — APPROVED** [review]

File: helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java (hunk 4 above)
Comment (kaisun2000) [reply helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java:78]: fixed.
